### PR TITLE
web: fix build-history-tab-switching bug

### LINKED ISSUE #4962: switching build tabs immediately after triggering a build is weird
## Bug Report

If you trigger a build and then switch tabs on the build page, you will automatically get switched back to the latest build but the URL will not update. After that you can't use the arrow keys to change tabs, and if you click tabs you will keep getting switched back to the latest build.

## Steps to Reproduce

1. Visit the build page
1. Trigger a build, either by clicking + or typing `T`
1. Quickly (like before the build prep finishes) switch tabs, either by clicking one or by typing `l`.

## Expected Results

* You see output for the tab you switched to
* Your URL reflects the tab you switched to

## Actual Results

* you get swapped back to the newly-triggered build
* your URL still says whatever tab you switched to

## Additional Context

I think this has to do with the `BuildFetched` callback, which updates the `id` and `name` fields on the build model and can cause them to go out of sync with the URL.

## Version Info

* Concourse version: 5.8.0 RC latest master
* Deployment type (BOSH/Docker/binary): docker-compose
* Infrastructure/IaaS: local
* Browser (if applicable): chrome
* Did this used to work? Yes, working in 5.7.2

### PR DESCRIPTION
**Pull request by pivotal-jamie-klassen**
# Existing Issue

Fixes #4962.

# Changes proposed in this pull request

The bug was observed in cases where you triggered a new build -- it turns out the important part of this flow was simply viewing a pending build. In such cases, the UI polls the GetBuild and GetBuildPrep endpoints with a 1-second delay. It is quite possible that, during this delay, the user will switch tabs and view a different build. Then, after the delay, the app will receive callbacks containing the build/prep for the pending build that was originally viewed. It is necessary to gracefully handle this data without displaying invalid or irrelevant data.

The fix ended up being to carefully ignore messages containing data from irrelevant builds - including build events being received and build/prep being fetched. This replaces the old `browsingIndex` method and has better (but not perfect) test coverage.

# Contributor Checklist
- [x] Unit tests
- [x] ~Integration tests (if applicable)~
- [x] ~Updated documentation (located at https://github.com/concourse/docs)~
- [x] ~Updated release notes (located at https://github.com/concourse/concourse/tree/master/release-notes)~

# Reviewer Checklist
- [ ] Code reviewed
- [ ] Tests reviewed
- [x] ~Documentation reviewed~
- [x] ~Release notes reviewed~
- [x] PR acceptance performed
- [x] ~New config flags added? Ensure that they are added to the [BOSH](https://github.com/concourse/concourse-bosh-release) and [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for the [integration tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env) (for example, if they are Garden configs that are not displayed in the `--help` text).~

### CODE REVIEW

File: web/elm/src/Build/Build.elm [hunk 1]
```diff
@@ -162,13 +162,14 @@ changeToBuild { highlight, pageType } ( model, effects ) =
...
 
 
 extractTitle : Model -> String
 extractTitle model =
-    case ( model.build, currentJob model, model.page ) of
-        ( RemoteData.Success build, Just { jobName }, _ ) ->
-            jobName ++ " #" ++ build.name
+    case ( model.hasLoadedYet, model.job, model.page ) of
+        ( True, Just { jobName }, _ ) ->
+            jobName ++ " #" ++ model.name
```
Comment (jomsie) [inline web/elm/src/Build/Build.elm]: i didn't realize the model always had a name, nice!

File: web/elm/src/Build/Build.elm [hunk 2]
```diff
@@ -209,7 +202,7 @@ handleCallback action ( model, effects ) =
                     else if status.code == 404 then
                         ( { model
                             | prep = Nothing
-                            , build = RemoteData.Failure err
+                            , notFound = True
```
Comment (jomsie) [inline web/elm/src/Build/Build.elm]: i really dig the `notFound` field on the model, this is super clear

**Comment by jomsie** [comment]
sidenote: am i crazy or does public/images/ic-terminal.svg not exist anywhere anymore?

File: web/elm/src/Build/Build.elm [hunk 3]
```diff
@@ -406,20 +396,14 @@ update msg ( model, effects ) =
...
 
         Click AbortBuildButton ->
-            (model.build
-                |> RemoteData.toMaybe
-                |> Maybe.map
-                    (.id >> DoAbortBuild >> (::) >> Tuple.mapSecond)
-                |> Maybe.withDefault identity
-            )
-                ( model, effects )
+            ( model, DoAbortBuild model.id :: effects )
```
Comment (jomsie) [inline web/elm/src/Build/Build.elm]: this is wonderful!

**Comment by kcmannem** [comment]
Everything works! Really cool feature, I didn't know about this or the `?` shortcut to discover all this.

**Comment by jomsie** [comment]
just reproduced and verified it works! gonna look it over once more.

**Pull request merged**